src/v/archival/ntp_archiver_service.cc

```diff
@@ -1340,19 +1340,16 @@ ss::future<ntp_archiver_upload_result> ntp_archiver::upload_segment(
     // manifest.
 }
 
-std::optional<ss::sstring> ntp_archiver::upload_should_abort() {
-    auto original_term = _parent.term();
-    auto lost_leadership = !_parent.is_leader()
-                           || _parent.term() != original_term;
-    if (unlikely(lost_leadership)) {
```

Comment:
>Fixes incorrect logic in `upload_should_abort()`,

please describe why the logic is incorrect, and what the new logic should be and why.

>and adds
`ntp_archiver::lost_leadership()` to abstract out `_parent` partition
leadership checking into its own function.

is there a relationship between lost_leadership and the bug fix?

a bug fix is a good example of a commit that can be broken out immediately upon discovery into a PR.